test/unit/plugins/providers/docker/provider_test.rb

```diff
@@ -127,5 +130,17 @@
 
       expect(subject.ssh_info).to eq(ssh_info)
     end
+
+    it "should receive a valid address when running with podman emulation" do
+      allow(provider_config).to receive(:compose).and_return(false)
+      allow(platform).to receive(:windows?).and_return(false)
+      allow(platform).to receive(:darwin?).and_return(false)
+      allow(driver_obj).to receive(:podman?).and_return(true)
+      allow(driver_obj).to receive(:created?).and_return(true)
+      allow(driver_obj).to receive(:state).and_return(:running)
+      allow(driver_obj).to receive(:inspect_container).and_return(podman_network_settings)
+
+      expect(subject.ssh_info).to eq({:host=> "127.0.0.1", :port => 2222})
+    end
```

Comment:
looks great! could you add a test to ensure that with podman emulation `nil` is returned if the port info is empty. It should be similar to https://github.com/hashicorp/vagrant/blob/7c1997e366317a79da0a90f6e641f04297497aa9/test/unit/plugins/providers/docker/provider_test.rb#L110-L120

one way to group the podman related tests is to use `context`, something like:
```
  context "podman emulation" do
    before do
      allow(driver_obj).to receive(:podman?).and_return(true)
    end
 
    it "should receive a valid address" do ..... end

    it "returns nil if a port info is nil from the drive" do ..... end
  end 
```

Reply:
you got it!